lib/modularity/extend.ex

```diff
@@ -334,26 +334,43 @@ defmodule Bonfire.Common.Extend do
     end
   end
 
+  @loaded_extensions_key {__MODULE__, :extensions_loaded}
   @decorate time()
   defp application_loaded?(extension) do
-    loaded_apps = loaded_applications()
-    app_names = Enum.map(loaded_apps, &elem(&1, 0))
+    extensions = :persistent_term.get(@loaded_extensions_key, %{})
 
-    Enum.member?(
-      app_names,
-      extension
-    )
+    case Map.get(extensions, extension) do
+      nil ->
+        result =
+          loaded_applications()
```

Comment:
doesn't this mean it will try `:persistent_term.get` a second time?

Reply:
oh I see it's a different key

Reply:
Yes but with a different key, to get the applications that are loaded.